File: src/ipc/nettools.ts
import { invoke } from "@tauri-apps/api/core";

export interface CommandOutput {
  stdout: string;
  stderr: string;
  exit_code: number;
}

export const ping = (target: string): Promise<CommandOutput> => invoke("net_ping", { target });
export const dnsLookup = (target: string): Promise<CommandOutput> => invoke("net_dns_lookup", { target });
export const arpTable = (): Promise<CommandOutput> => invoke("net_arp_table");
export const portCheck = (target: string, port: number): Promise<CommandOutput> =>
  invoke("net_port_check", { target, port });
export const traceroute = (target: string): Promise<CommandOutput> => invoke("net_traceroute", { target });

File: src-tauri/src/commands/nettools.rs
use crate::core::nettools::{self, CommandOutput};

#[tauri::command]
pub async fn net_ping(target: String) -> Result<CommandOutput, String> {
    nettools::ping(&target).await.map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn net_dns_lookup(target: String) -> Result<CommandOutput, String> {
    nettools::dns_lookup(&target).await.map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn net_arp_table() -> Result<CommandOutput, String> {
    nettools::arp_table().await.map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn net_port_check(target: String, port: u16) -> Result<CommandOutput, String> {
    nettools::port_check(&target, port).await.map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn net_traceroute(target: String) -> Result<CommandOutput, String> {
    nettools::traceroute(&target).await.map_err(|e| e.to_string())
}
